web/src/lib/components/album-page/album-viewer.svelte

```diff
@@ -32,14 +32,57 @@
 
   dragAndDropFilesStore.subscribe((value) => {
     if (value.isDragging && value.files.length > 0) {
-      handlePromiseError(fileUploadHandler(value.files, album.id));
+      if (isExternalUser()) {
+        if (
+          confirm(
+            'Please note uploaded files can only be deleted by Immich users. Please upload at your own risk. Are you sure you want to proceed?',
+          )
+        ) {
+          handlePromiseError(fileUploadHandler(value.files, album.id));
+        }
+      } else {
+        handlePromiseError(fileUploadHandler(value.files, album.id));
+      }
       dragAndDropFilesStore.set({ isDragging: false, files: [] });
     }
   });
 
   const downloadAlbum = async () => {
     await downloadArchive(`${album.albumName}.zip`, { albumId: album.id });
   };
+
+  const handleDrop = (event: DragEvent) => {
```

Comment:
Why are this and the handleDrag functions needed?

Reply:
This is to handle when a user drops a file onto the screen instead of using the upload button at the top. Not sure if there is a better way to handle this?

Reply:
This case is already handled with `dragAndDropFilesStore` and all this seems redundant to me

Reply:
Feel free to submit changes mate, I am out for the next couple of days so can't action anything 👍